react/features/recording/middleware.js

```diff
@@ -0,0 +1,39 @@
+/* @flow */
+
+import { CONFERENCE_WILL_JOIN } from '../base/conference';
+import { JitsiConferenceEvents } from '../base/lib-jitsi-meet';
+import { MiddlewareRegistry } from '../base/redux';
+
+import { updateRecordingSessionData } from './actions';
+
+/**
+ * The redux middleware to handle the recorder updates in a React way.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(({ dispatch }) => next => action => {
+    const result = next(action);
+
+    switch (action.type) {
+    case CONFERENCE_WILL_JOIN: {
+        const { conference } = action;
+
+        conference.on(
+            JitsiConferenceEvents.RECORDER_STATE_CHANGED,
```

Comment:
I think this belongs here: https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/conference/actions.js#L69

Reply:
I started to move it to what you suggested and then I realised that it would mean that a base feature would reference a non-base feature which I generally don't like. We also talked about this with Lyubo (since we just had our catchup while I was working on it) and he also said that recording can easily remain a black box and keep its event handling in there. What do you think?